library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,890 @@
+        'description',
+        'sourcePort',
+        'translateAddress',
+        'translatePort',
+        'vlansDisabled',
+        'vlansEnabled',
+        'vlans',
+        'rules',
+        'autoLasthop',
+        'pool',
+        'lastHopPool',
+        'fallbackPersistence',
+        'ipProtocol',
+        'mask',
+        'disabled',
+        'enabled'
+    ]
+
+    returnables = [
+        'description',
+        'disabled',
+        'enabled',
+        'ip_protocol',
+        'mask',
+        'address',
+        'disabled_vlans',
+        'enabled_vlans',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'translate_address',
+        'translate_port',
+        'fallback_persistence',
+        'source_port',
+        'vlans',
+        'pool',
+        'last_hop_pool',
+        'irules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    updatables = [
+        'description',
+        'disabled',
+        'enabled',
+        'mask',
+        'address',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'disabled_vlans',
+        'enabled_vlans',
+        'state',
+        'ip_protocol',
+        'fallback_persistence',
+        'translate_address',
+        'translate_port',
+        'source_port',
+        'vlans',
+        'pool',
+        'last_hop_pool',
+        'irules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    @property
+    def state(self):
+        if self._values['state'] == 'enabled':
+            return 'present'
+        return self._values['state']
+
+    @property
+    def enabled(self):
+        if self._values['enabled'] is None:
+            return None
+        return True
+
+    @property
+    def disabled(self):
+        if self._values['disabled'] is None:
+            return None
+        return True
+
+
+class ApiParameters(Parameters):
+    @property
+    def irules(self):
+        if self._values['irules'] is None:
+            return []
+        return self._values['irules']
+
+
+class ModuleParameters(Parameters):
+    @property
+    def description(self):
+        if self._values['description'] is None:
+            return None
+        elif self._values['description'] in ['none', '']:
+            return ''
+        return self._values['description']
+
+    @property
+    def translate_address(self):
+        if self._values['translate_address'] is None:
+            return None
+        if self._values['translate_address']:
+            return 'enabled'
+        else:
+            return 'disabled'
+
+    @property
+    def translate_port(self):
+        if self._values['translate_port'] is None:
+            return None
+        if self._values['translate_port']:
+            return 'enabled'
+        else:
+            return 'disabled'
+
+    @property
+    def advertise(self):
+        if self._values['advertise'] is None:
+            return None
+        return flatten_boolean(self._values['advertise'])
```

Comment:
does API require yes/no ? if yes then you can leave it, if not change accordingly

Reply:
Yes - the API sets advertise field to yes/no; by default, it is set to no

```
    "address": "192.0.2.0",
    "advertise": "no",
    "autoLasthop": "default",
```

Reply:
This comment is addressed now.